### PR TITLE
server: Switch to echo

### PR DESCRIPTION
**Pull request by Gundersanne**

### CODE REVIEW

File: internal/server/server.go [hunk 1]
```diff
@@ -13,161 +13,116 @@ import (
...
 
 type Handlers struct{}
 
-func (s *Handlers) GetVersion(w http.ResponseWriter, r *http.Request) {
+func (s *Handlers) GetVersion(ctx echo.Context) error {
 	spec, err := GetSwagger()
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
+		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
```
Comment (larskarlitski) [inline internal/server/server.go:24]: Just returning `err` is enough here. Echo turns non`HTTPError`-errors into internal server errors automatically.

File: internal/server/server.go [hunk 2]
```diff
@@ -13,161 +13,116 @@ import (
...
 	}
 
-	decoder := json.NewDecoder(r.Body)
+	decoder := json.NewDecoder(ctx.Request().Body)
 	var composeRequest cloudapi.ComposeJSONRequestBody
 	err := decoder.Decode(&composeRequest)
 	if err != nil {
-		http.Error(w, fmt.Sprintf("Failed decoding the compose request %s", err.Error()), http.StatusBadRequest)
-		return
+		return echo.NewHTTPError(http.StatusBadRequest, "Failed decoding the compose request %s", err.Error())
```
Comment (larskarlitski) [inline internal/server/server.go:85]: `NewHTTPError()` does not accept format strings. (We are all making this mistake independently.)

Please use `fmt.Sprintf()`. Also, when using the `%v` directive, you don't need to call the `Error()` method:
```suggestion
		return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Failed decoding the compose request %v", err))
```

**Review by larskarlitski — CHANGES_REQUESTED** [review]
Thank you! Two small comments, which might need fixing in a few more places than the ones I pointed them to.

Otherwise this is perfect.

File: internal/server/server.go (hunk 1 above)
Comment (Gundersanne) [reply internal/server/server.go]: neat!

File: internal/server/server.go (hunk 2 above)
Comment (Gundersanne) [reply internal/server/server.go]: I've been using "" + err.Error(); "%v", error, "%s", err.Error(); fmt() or not intermittently, good to finally know how to do it :D

**Review by larskarlitski — APPROVED** [review]
Thanks for the quick turnaround!

**Pull request merged**